crates/feldera-storage/src/backend/mod.rs

```diff
@@ -0,0 +1,244 @@
+//! Storage backend for Feldera.
+
+use std::ops::{Deref, DerefMut};
+
+#[cfg(feature = "glommio")]
+use glommio::io::{DmaBuffer, ReadResult};
+use thiserror::Error;
+
+#[cfg(feature = "glommio")]
+pub mod glomm;
+pub mod monoio;
+
+#[cfg(test)]
+mod tests;
+
+/// A file-descriptor we can write to.
+pub struct FileHandle(i64);
+
+/// A file-descriptor we can read or prefetch from.
+pub struct ImmutableFileHandle(i64);
+
+/// An implementation detail that abstracts different write buffers types
+/// required by different backends.
+///
+/// Note: If we decide to e.g., stick with only monoio we can avoid this
+/// abstraction.
+enum InnerWriteBuffer {
+    #[cfg(feature = "glommio")]
+    Glommio(DmaBuffer),
+    Vec(Vec<u8>),
+}
+
+impl InnerWriteBuffer {
+    #[cfg(feature = "glommio")]
+    fn unwrap_as_dma_buffer(self) -> DmaBuffer {
+        match self {
+            InnerWriteBuffer::Glommio(x) => x,
+            InnerWriteBuffer::Vec(_) => panic!("unwrap called on invalid buffer"),
+        }
+    }
+
+    fn unwrap_as_vec(self) -> Vec<u8> {
+        match self {
+            InnerWriteBuffer::Vec(x) => x,
+            _ => panic!("unwrap called on invalid buffer"),
+        }
+    }
+
+    fn len(&self) -> usize {
+        match self {
+            #[cfg(feature = "glommio")]
+            InnerWriteBuffer::Glommio(x) => x.len(),
+            InnerWriteBuffer::Vec(x) => x.len(),
+        }
+    }
+}
+
+/// A buffer suitable for writing to a file using Direct I/O over io_uring.
```

Comment:
Can we get an alignment guarantee for the inner data? `rkyv` wants at least 16 bytes.